### PR TITLE
clean up status transition

### PR DESCRIPTION
**Pull request by ticoann**
@amaltaro, Alan please review. Mostly for preventing move to failed status automatically

### CODE REVIEW

File: src/python/WMCore/RequestManager/RequestDB/Settings/RequestStatus.py [hunk 1]
```diff
@@ -35,19 +30,14 @@
...
- "running-open" : ["running-closed", "aborted", "failed", "epic-FAILED"],
- "running-closed" : ["force-complete", "completed", "aborted", "failed", "epic-FAILED"],
+ "acquired" : ["running-open", "completed", "acquired", "aborted"],
+ "running" : ["completed", "aborted"],
+ "running-open" : ["running-closed", "aborted"],
+ "running-closed" : ["force-complete", "completed", "aborted"],
  "force-complete" : ["completed"],
- "failed" : ["failed", "testing-approved", "assigned", "rejected"],
- "epic-FAILED" : ["epic-FAILED"],
+ "failed" : ["failed", "assigned", "rejected"],
```
Comment (amaltaro) [inline src/python/WMCore/RequestManager/RequestDB/Settings/RequestStatus.py]: how about we remove "failed" from "failed" status?

File: src/python/WMCore/RequestManager/RequestDB/Settings/RequestStatus.py [hunk 2]
```diff
@@ -35,19 +30,14 @@
...
- "testing" : ["testing", "tested", "test-failed", "aborted"],
- "tested" : ["tested", "assignment-approved","failed", "rejected", "aborted"],
- "test-failed" : ["test-failed", "testing-approved", "rejected", "aborted"],
  "assignment-approved" : ["assignment-approved", "assigned", "rejected"],
  "assigned" : ["assigned","acquired", "aborted", "failed"],
- "acquired" : ["running-open", "failed", "completed", "acquired", "aborted"],
- "running" : ["completed", "aborted", "failed", "epic-FAILED"],
- "running-open" : ["running-closed", "aborted", "failed", "epic-FAILED"],
- "running-closed" : ["force-complete", "completed", "aborted", "failed", "epic-FAILED"],
+ "acquired" : ["running-open", "completed", "acquired", "aborted"],
```
Comment (amaltaro) [inline src/python/WMCore/RequestManager/RequestDB/Settings/RequestStatus.py]: Hmmm, I'm not really sure whether this is a legit use case... Can't an "acquired" workflow move to "failed" according to its input or some other constraint?

**Comment by amaltaro** [comment]
Seangchan, do you know whether this state transition is done on GQ or on the agent side?
If it's on the agent side, then we'll be able to debug it in the next time it happen.

File: src/python/WMCore/RequestManager/RequestDB/Settings/RequestStatus.py (hunk 1 above)
Comment (ticoann) [inline src/python/WMCore/RequestManager/RequestDB/Settings/RequestStatus.py]: I remember there was some problem doing that in ReqMgr2. I am not sure what it was. But for the safe side I left it there.

File: src/python/WMCore/RequestManager/RequestDB/Settings/RequestStatus.py (hunk 2 above)
Comment (ticoann) [inline src/python/WMCore/RequestManager/RequestDB/Settings/RequestStatus.py]: I don't think anybody move the failed status manually. I think Ops use only reject an abort. Failed status moves automatically if the work is not pulled for a long time by workqueue, The problem is there is the case when only partial work is done and other element got stuck somewhere it still have possibility moving to failed status automatically

**Comment by ticoann** [comment]
Alan, I will merge this and see whether it causes any further problem I haven't expected.

**Pull request merged**